core/src/main/scala/org/apache/spark/deploy/LocalSparkCluster.scala

```diff
@@ -66,6 +69,9 @@ class LocalSparkCluster(
       val workDir = if (Utils.isTesting) {
         Utils.createTempDir(namePrefix = "worker").getAbsolutePath
       } else null
+      if (workDir != null) {
```

Comment:
nit: shall we check `Utils.isTesting` here?

Reply:
done